vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/application_code/nuvoton_code/entropy_hardware_poll.c

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Hardware entropy collector for M487 RNG
+ *
+ */
+
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include "FreeRTOS.h"
+#include "task.h"
+#include "NuMicro.h"
+
+/*
+ * Get Random number generator.
+ */
+ #define PRNG_KEY_SIZE  (0x20UL)
+ 
+static volatile int  g_PRNG_done;
+volatile int  g_AES_done;
+
+
+#define SNUM        32       /* recorded number of lastest samples */
+
+
+static uint32_t   adc_val[SNUM];
+static uint32_t   val_sum;
+static int        oldest;
+
+#ifdef __ICCARM__
+#define __inline   inline
+#endif
+
+static __inline uint32_t  get_adc_bg_val()
+{
+    EADC->SWTRG = (1 << 16);      //Trigger Band-gap
+    while ((EADC->STATUS1 & 1) != 1);
+    return (EADC->DAT[16] & 0xFFF);
+}
+
+int  adc_trng_gen_bit()
+{
+    uint32_t   new_val, average;
+    int        ret_val;
+    int ref_bg_ub = ((1.2/3.3)*4095) + 50;
```

Comment:
There's no explanation of where these values came from.

Reply:
@dcgaws ,  I have no write permission on this PR's branch, so pushed one commit 918e6ed443e for the explanation at: [my repo](https://github.com/cyliangtw/amazon-freertos/tree/feature/m487_cmake_0531).